lib/sisimai/bite/email/mxlogic.rb

```diff
@@ -0,0 +1,253 @@
+    # Sisimai::Bite::Email::MXLogic parses a bounce email which created by
+    # McAfee SaaS (formerly MX Logic).
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/MXLogic.pm
+      # Based on Sisimai::Bite::Email::Exim
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :'from'      => %r/\AMail Delivery System/,
+        :'subject'   => %r{(?:
+             Mail[ ]delivery[ ]failed(:[ ]returning[ ]message[ ]to[ ]sender)?
+            |Warning:[ ]message[ ].+[ ]delayed[ ]+
+            |Delivery[ ]Status[ ]Notification
+            )
+        }x,
+        :'message-id' => %r/\A[<]mxl[~][0-9a-f]+/,
+      }
+      Re1 = {
+        :rfc822 => %r/\AIncluded is a copy of the message header:\z/,
+        :begin  => %r/\AThis message was created automatically by mail delivery software[.]\z/,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReCommand = [
+        %r/SMTP error from remote (?:mail server|mailer) after ([A-Za-z]{4})/,
+        %r/SMTP error from remote (?:mail server|mailer) after end of ([A-Za-z]{4})/,
+      ]
+      ReFailure = {
+        userunknown: %r{
+            user[ ]not[ ]found
+        }x,
+        hostunknown: %r{(?>
+             all[ ](?:
+                 host[ ]address[ ]lookups[ ]failed[ ]permanently
+                |relevant[ ]MX[ ]records[ ]point[ ]to[ ]non[-]existent[ ]hosts
+                )
+            |Unrouteable[ ]address
+            )
+        }x,
+        mailboxfull: %r{(?:
+             mailbox[ ]is[ ]full:?
+            |error:[ ]quota[ ]exceed
+            )
+        }x,
+        notaccept: %r{(?:
+             an[ ]MX[ ]or[ ]SRV[ ]record[ ]indicated[ ]no[ ]SMTP[ ]service
+            |no[ ]host[ ]found[ ]for[ ]existing[ ]SMTP[ ]connection
+            )
+        }x,
+        systemerror: %r{(?>
+             delivery[ ]to[ ](?:file|pipe)[ ]forbidden
+            |local[ ]delivery[ ]failed
+            |LMTP[ ]error[ ]after[ ]
+            )
+        }x,
+        contenterror: %r{
+            Too[ ]many[ ]["]Received["][ ]headers
+        }x,
+      }
+      ReDelayed = %r{(?:
+         retry[ ]timeout[ ]exceeded
+        |No[ ]action[ ]is[ ]required[ ]on[ ]your[ ]part
+        |retry[ ]time[ ]not[ ]reached[ ]for[ ]any[ ]host[ ]after[ ]a[ ]long[ ]failure[ ]period
+        |all[ ]hosts[ ]have[ ]been[ ]failing[ ]for[ ]a[ ]long[ ]time[ ]and[ ]were[ ]last[ ]tried
+        |Delay[ ]reason:[ ]
+        |Message[ ].+[ ](?:has[ ]been[ ]frozen|was[ ]frozen[ ]on[ ]arrival[ ]by[ ])
+        )
+      }x
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'McAfee SaaS'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      # X-MX-Bounce: mta/src/queue/bounce
+      # X-MXL-NoteHash: ffffffffffffffff-0000000000000000000000000000000000000000
+      # X-MXL-Hash: 4c9d4d411993da17-bbd4212b6c887f6c23bab7db4bd87ef5edc00758
```

Comment:
__[rubocop]__
* Style/Semicolon: Do not use semicolons to terminate expressions. [:ref](https://github.com/bbatsov/ruby-style-guide#no-semicolon)
* Layout/TrailingWhitespace: Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

<!-- Comment by [SideCI](https://sideci.com) -->